testutil/keeper/proof.go

```diff
@@ -11,29 +11,42 @@ import (
 	cmtproto "github.com/cometbft/cometbft/proto/tendermint/types"
 	dbm "github.com/cosmos/cosmos-db"
 	"github.com/cosmos/cosmos-sdk/codec"
+	addresscodec "github.com/cosmos/cosmos-sdk/codec/address"
 	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
 	"github.com/cosmos/cosmos-sdk/runtime"
+	"github.com/cosmos/cosmos-sdk/testutil/integration"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
 	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
 	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
 	"github.com/golang/mock/gomock"
 	"github.com/stretchr/testify/require"
 
-	"github.com/pokt-network/poktroll/testutil/proof"
+	"github.com/pokt-network/poktroll/app"
 	"github.com/pokt-network/poktroll/testutil/proof/mocks"
+	appkeeper "github.com/pokt-network/poktroll/x/application/keeper"
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
+	gatewaykeeper "github.com/pokt-network/poktroll/x/gateway/keeper"
+	gatewaytypes "github.com/pokt-network/poktroll/x/gateway/types"
 	"github.com/pokt-network/poktroll/x/proof/keeper"
 	"github.com/pokt-network/poktroll/x/proof/types"
+	prooftypes "github.com/pokt-network/poktroll/x/proof/types"
+	sessionkeeper "github.com/pokt-network/poktroll/x/session/keeper"
 	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+	supplierkeeper "github.com/pokt-network/poktroll/x/supplier/keeper"
+	suppliertypes "github.com/pokt-network/poktroll/x/supplier/types"
 )
 
-func ProofKeeper(
-	t testing.TB,
-	sessionByAppAddr proof.SessionsByAppAddress,
-) (keeper.Keeper, context.Context) {
-	t.Helper()
+type ProofKeeperWithDeps struct {
+	ProofKeeper       *keeper.Keeper
+	SessionKeeper     prooftypes.SessionKeeper
+	SupplierKeeper    prooftypes.SupplierKeeper
+	ApplicationKeeper prooftypes.ApplicationKeeper
+}
 
+func ProofKeeper(t testing.TB) (keeper.Keeper, context.Context) {
```

Comment:
We have a `ProofKeeper` function and a `ProofKeeperWithDeps` function.

I feel like it should be one of:
- `ProofKeeper` and `ProofKeeperWthDeps`
- `NewProofKeeper` and `NewProofKeeperWithDeps`

Reply:
I like the name and approach!